src/chart/generator/marker.cpp

```diff
@@ -88,7 +94,12 @@ Marker::Marker(const Options &options,
 	    index,
 	    !horizontal ? &mainId->value : subAxisId);
 
-	spacing.y = (!horizontal || lineOrCircle)
+	spacing.y = (!horizontal || lineOrCircle
+	                || (channels.at(ChannelId::x).isDimension()
+	                    && channels.at(ChannelId::x).hasDimension()
```

Comment:
This expression is getting too complex. Maybe extract newly added portion to a local variable?